### PR TITLE
Completed task 8 - PATCH /api/articles/article_id

### PR DESCRIPTION
**Pull request by Clatherine**
Added endpoint for PATCH /api/articles/article_id
added associated tests
added endpoint to endpoints.json

### CODE REVIEW

**Comment by Clatherine** [comment]
Note: final test rejects a request that includes additional properties on the send body, but maybe these should just be ignored, not sure!

File: models/articles.model.js [hunk 1]
```diff
@@ -29,5 +29,27 @@ exports.checkArticleIdExists = (article_id)=>{
...
\ No newline at end of file
+}
+
+exports.editArticle = (article_id, body)=>{
+    return db.query("SELECT votes FROM articles WHERE article_id = $1", [article_id]).then(({rows})=>{
+      const currentVotes = rows[0].votes 
+      const increment = body.inc_votes
+      const updatedVotes = currentVotes + increment
+      if(updatedVotes<0){
+        return Promise.reject({status: 400, msg:'We\'re not popular enough to subtract that amount!'})
```
Comment (Harrison365) [inline models/articles.model.js:42]: Clever 👍

File: models/articles.model.js [hunk 2]
```diff
@@ -29,5 +29,27 @@ exports.checkArticleIdExists = (article_id)=>{
...
             return Promise.reject({status: 404, msg:"That article does not exist!"})
             }
+    
     })
-}
\ No newline at end of file
+}
+
+exports.editArticle = (article_id, body)=>{
+    return db.query("SELECT votes FROM articles WHERE article_id = $1", [article_id]).then(({rows})=>{
```
Comment (Harrison365) [inline models/articles.model.js:37]: You shouldn't need to do this. You can simply update the votes to be the current votes + new votes

File: models/articles.model.js [hunk 3]
```diff
@@ -29,5 +29,27 @@ exports.checkArticleIdExists = (article_id)=>{
...
+      const updatedVotes = currentVotes + increment
+      if(updatedVotes<0){
+        return Promise.reject({status: 400, msg:'We\'re not popular enough to subtract that amount!'})
+      }
+      else {
+      return db.query("UPDATE articles SET votes = $1 WHERE article_id = $2 RETURNING *", [updatedVotes, article_id])
+    .then(({rows})=>{
+        if(rows.length===0){
+            return Promise.reject({status: 404, msg:"That article does not exist!"})
+            }
```
Comment (Harrison365) [inline models/articles.model.js:49]: Does this not make your checkArticleIdExists(article_id) in the controller redundant?

File: __tests__/integration-test.js [hunk 4]
```diff
@@ -238,3 +239,131 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+        .send({ inc_votes : 1,
+            author: "Newton", 
+            inc_votes: 3
+        })
+        .expect(400)
+        .then(({body})=>{
+          expect(body.msg).toBe('Unexpected properties on request body')
+        })
+    })
+})
```
[the remark this answers is not in the document]
Comment (Clatherine) [reply __tests__/integration-test.js:369]: Added this in!

File: models/articles.model.js (hunk 3 above)
Comment (Clatherine) [reply models/articles.model.js:49]: Yep! Have deleted this repetition now

File: models/articles.model.js (hunk 2 above)
Comment (Clatherine) [reply models/articles.model.js:37]: Agreed to leave as is so can prevent negative votes.

**Pull request merged**